app/svelte/src/server/svelte-stories-loader.ts

```diff
@@ -0,0 +1,21 @@
+import dedent from 'ts-dedent';
+import { readFileSync } from 'fs';
+
+import { extractStoriesSources } from './extract-sources';
+
+const parser = require.resolve('../client/parse-stories').replace(/[/\\]/g, '/');
+
+function transformSvelteStories(code: string) {
+  // eslint-disable-next-line no-underscore-dangle
+  const { resource } = this._module;
+
+  const source = readFileSync(resource).toString();
+  const storiesSources = extractStoriesSources(source);
+
+  return dedent`${code}
+    const { default: parser } = require('${parser}');
+    export const __dynamicStoriesGenerator = (m) => parser(m, ${JSON.stringify(storiesSources)});
```

Comment:
Can you explain what's happening here? It looks like it is generating the story list at runtime rather than at this point (build time). It would be (much) better if it was able to generate a story list statically!

Reply:
I'll try to explain why I had to do this :)

Generating native Js stories from a svelte file can be hard:

```
<Story name="myStoryName" args={ <can be complex javascript expression > }/>
```

In this example, `name` is static, but any other properties can be complex expressions (for example by referencing common properties, etc). To generate a js story from this definition, I had to implement an ast->js generator and this can be hard to maintain and will depends on svelte internals.

So I choose to not follow this path and "execute" the stories in a special environment which will just collect every stories definition/properties, and disable components rendering. This is the `parser` function you see here.

At the end, I have a map of stories id to stories definition.

I can export them with `module.exports = ..`. 

However the svelte component is compiled as esm, so this syntax is not possible. I use it in the jest-transformer, because jest requires cjs module and in this context, svelte are compiled as cjs during tests. I don't want to require svelte files to be compiled to cjs because it will be a breaking change (and cjs is not really a step forward...)

I can't export them as named export because this map of stories is dynamic and named export should be statically defined.

A scenario I tried is introspecting the stories component to extract all stories name (I already do this to extract stories sources.), and for each stories, I can generate a named `export`. However, I can't introspect easily "imported stories" (story defined in another file and imported in another file). I'd like to support this feature because it allows to reuse stories in different contexts (there is a paragraph about this in the doc).

So I decided to use this "trick" : I export a special function which is used in `loadCsf`: if this function is defined, then `loadCsf` use the result instead of the module.

I knew I would have remarks on this change, it's the only change external to svelte - but I wanted to have your opinion on this.

Reply:
> A scenario I tried is introspecting the stories component to extract all stories name (I already do this to extract stories sources.), and for each stories, I can generate a named export. However, I can't introspect easily "imported stories" (story defined in another file and imported in another file). I'd like to support this feature because it allows to reuse stories in different contexts (there is a paragraph about this in the doc).

Can you expand on this part?

I think this is the only piece blocking doing everything in the loader and not modifying `loadCsf`, which we don't think is a good idea since it's important that CSF exports be statically analyzable.

Reply:
I'd have liked to be able to compose stories, like described somewhere in the documentation, but I can't find the page I though of. I will try to do without.

> it's important that CSF exports be statically analyzable.

Can I ask why ? In the case of React, a story is a real react function, but In my case, nothing is reusable/importable.